Bottom: 69c655994e71af522643f99474add76742e8cc73
Top:    263f2fbbbfc687ee902ccd5da4226645a11a9065
Author: Kevin Redon <kevredon@mail.tsaitgaist.info>
Date:   2012-12-07 15:31:58 +0100

Don't merge lines with different clearances

When multiple lines (with same direction, thickness, and clear flags)
are moved, they are merged into one line. We should also check the
line clearances match to ensure the lines are suitable for merging.

[Commit message re-worded by Peter Clifton]

Signed-off-by: Kevin Redon <kevredon@mail.tsaitgaist.info>


---

diff --git a/src/create.c b/src/create.c
index b39ff65..7919b43 100644
--- a/src/create.c
+++ b/src/create.c
@@ -301,6 +301,7 @@ struct line_info
 {
   Coord X1, X2, Y1, Y2;
   Coord Thickness;
+  Coord Clearance;
   FlagType Flags;
   LineType test, *ans;
   jmp_buf env;
@@ -335,9 +336,11 @@ line_callback (const BoxType * b, void *cl)
       longjmp (i->env, 1);
     }
   /* remove unnecessary line points */
-  if (line->Thickness == i->Thickness
+  if (line->Thickness == i->Thickness &&
+      /* don't merge lines if the clearances differ  */
+      line->Clearance == i->Clearance &&
       /* don't merge lines if the clear flags differ  */
-      && TEST_FLAG (CLEARLINEFLAG, line) == TEST_FLAG (CLEARLINEFLAG, i))
+      TEST_FLAG (CLEARLINEFLAG, line) == TEST_FLAG (CLEARLINEFLAG, i))
     {
       if (line->Point1.X == i->X1 && line->Point1.Y == i->Y1)
 	{
@@ -418,6 +421,7 @@ CreateDrawnLineOnLayer (LayerType *Layer,
   info.Y1 = Y1;
   info.Y2 = Y2;
   info.Thickness = Thickness;
+  info.Clearance = Clearance;
   info.Flags = Flags;
   info.test.Thickness = 0;
   info.test.Flags = NoFlags ();
